fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/NestedField.java

```diff
@@ -116,8 +116,8 @@ public int hashCode() {
     }
 
     @Override
-    public int planHash() {
-        return super.planHash() + getChild().planHash();
+    public int planHash(PlanHashKind hashKind) {
```

Comment:
These combinations are not good. This should be `x * 31 + y` or something like that. I wonder if we can do something about this as changing this would definitely break a lot of plan hashes.